### ISSUE TITLE
Host dev branch using vercel

### ISSUE THREAD

**Issue by IceSentry** [bevyengine/bevy-website#313]
We want to be able to host a dev version of the website to make collaboration easier. The main issue stopping us is that github pages can only host a single branch.

Fortunately, there's another way. We could use [vercel](https://vercel.com)/ to host multiple branches for free and it's super easy to setup.

**Comment by CooCooCaCha** [comment]
A few notes from testing the vercel <-> github integration on my fork.

- Vercel will automatically detect branch updates and deploy that branch to a unique (and ugly) url.
- Vercel will automatically comment on PRs with the generated url. I believe Vercel can be configured to not submit this comment.
- If a PR is updated, Vercel will redeploy and edit the original comment if needed.
- These urls will not be indexed by search engines. This is done by using the `x-robots-tag: noindex` HTTP header (https://vercel.com/support/articles/are-vercel-preview-deployment-indexed-by-search-engines#x-robots-tag-header).

Here's a screenshot of what the above looks like:
![vercel_github_screenshot](https://user-images.githubusercontent.com/850228/161401757-0b6578fc-7656-4280-ae7d-8bda4d94cd88.png)

**Comment by alice-i-cecile** [comment]
Cart and I have been chatting with Vercel's sales team: they'd be open to offering us the equivalent of their pro plan if we're willing to credit them for their hosting. The guidelines are [here](https://vercel.com/support/articles/can-vercel-sponsor-my-open-source-project), but we're discussing to see if we can compromise on something more consistent with the other sponsorships.

**Comment by alice-i-cecile** [comment]
So, for acknowledgement, we need:

1. A Infrastructure Sponsors section of our Sponsors section.
2. A link in the footer with the Vercel logo and OSS link (as described above), with the phrase "Website powered by Vercel".
3. A section in the README of the `bevy-website` repo for sponsors (with all of our Bronze+ tier sponsors).

This is ready for a test PR enabling Vercel integration + the acknowledgements, which we can then send to their sales team for approval.

**Comment by cart** [comment]
> 2. A link in the footer with the Vercel logo and OSS link (as described above), with the phrase "Website powered by Vercel".

I think we should try pushing back on this, given that the website isn't really "powered" by it. We're using it as a dev tool to improve our PR efficiency, not as a technology stack for the website (or as a host for the production website).

**Comment by alice-i-cecile** [comment]
> I think we should try pushing back on this, given that the website isn't really "powered" by it. We're using it as a dev tool to improve our PR efficiency, not as a technology stack for the website (or as a host for the production website).

Yeah, fair 🤔 We could enable it only for the vercel deployed branches I guess?

**Comment by cart** [comment]
That sits better with me, but thats a context that accomplishes goals different from what I think they're trying to accomplish. For simplicity, I'd prefer to push back wholesale first.

**Comment by cart** [comment]
Just sent them an email pushing back. We'll see how it plays out!

**Comment by alice-i-cecile** [comment]
> Sure, I get where you're coming from. It'll take longer to approve this since it doesn't conform to our sponsorship requirements, but if you send over a preview that has an infrastructure sponsors section added I can see if it's possible to get that approved.

Alright! Let's put something together. We just need a draft PR with:
- Vercel support enabled
- a Infrastructures Sponsor section with the Vercel logo and the corresponding link 🎉

**Comment by alice-i-cecile** [comment]
Still no word on the permissions requirements. We should start talking to competitors IMO.

**Comment by CooCooCaCha** [comment]
> Still no word on the permissions requirements. We should start talking to competitors IMO.

I have a branch that adapts the Vercel scripts to Netlify. Not saying we have to use Netlify but many of these services are so similar that it shouldn’t be too difficult to adapt.

So far it seems that Netlify and Cloudflare are potential options. Netlify sponsors Amethyst so perhaps they’ll sponsor Bevy.

**Comment by CooCooCaCha** [comment]
Meant to link my branch 

https://github.com/CooCooCaCha/bevy-website/pull/1

**Comment by alice-i-cecile** [comment]
Yeah, I'm on board with Netlify. Ideally we can just use their free tier?

**Comment by alice-i-cecile** [comment]
@IceSentry can you edit this issue title and description to something more general?

**Comment by MyZeD** [comment]
Vercel has an old architecture, even though they're "relatively" new to the market.
No IPv6, even though it's been standard since 2017: https://vercel.com/docs/domains/troubleshooting#ipv6-support

From me, a resounding: No.

Stay on Github Pages or search for other Provider.
